include/grpcpp/impl/codegen/proto_utils.h

```diff
@@ -96,9 +96,17 @@ Status GenericDeserialize(ByteBuffer* buffer, grpc::protobuf::Message* msg) {
   return result;
 }
 
-// this is needed so the following class does not conflict with protobuf
+// This is needed so the following class does not conflict with protobuf
 // serializers that utilize internal-only tools.
 #ifdef GRPC_OPEN_SOURCE_PROTO
+
+// These allow other Google projects that exist internally and in OSS (cough 
+// cough TensorFlow) to utilize certain internal optimization (cough cough 
+// Cord) that should be open sourced soon (sniff sneeze Abseil) clearing this
+// whole mess up.
+typedef GrpcProtoBufferWriter GrpcMaybeCordProtoBufferWriter;
+typedef GrpcProtoBufferWriter GrpcMaybeCordProtoBufferReader;
```

Comment:
So this is just a typedef for the convenience of the application program, correct? If that's the case, why doesn't the application program just do this #ifdef and typedef themselves? Otherwise I'm afraid that we might end up maintaining other typedefs and ifdefs for others.

Reply:
This entire block is already `ifdef`-ed out in google3 to support teams seamlessly picking up internal version of SerializationTraits. And I figure we could do the same for the  ProtoBuffer helper classes. I am sure that they have their own #defines, but I feel like this is something that validly falls in our court.

It's a little more than convenience IMO. TF is the first use case of a google3/OSS project using more than SerializationTraits, since they now use these helper classes. These shims let TF just use `GrpcMaybeCordProtoBufferWriter`. Then in OSS, it picks up `GrpcProtoBufferWriter`, an in google3 (via internal only files), they will pick up the Cord version of the class. I think since we support this opacity with SerializationTraits, we can with the helper classes too.